src/cargo/core/resolver/errors.rs

```diff
@@ -246,16 +246,23 @@ pub(super) fn activation_error(
             let mut names = candidates
                 .iter()
                 .take(3)
-                .map(|c| c.1.as_str())
+                .map(|c| "`".to_owned() + c.1.as_str() + "`")
                 .collect::<Vec<_>>();
-
+q
             if candidates.len() > 3 {
-                names.push("...");
+                names.push("...".to_owned());
             }
 
-            msg.push_str("did you mean: ");
-            msg.push_str(&names.join(", "));
-            msg.push_str("\n");
+            msg.push_str("perhaps you meant: ");
+            msg.push_str(&names.iter().enumerate().fold(
+                String::default(),
+                |acc, (i, el)| match i {
+                    0 => acc + el,
+                    i if names.len() - 1 == i && candidates.len() <= 3 => acc + " or " + el,
+                    _ => acc + ", " + el,
```

Comment:
Is this idiomatic way to do this?

Reply:
Not sure, but it looks like how I'd do it :)